tests/RobotFramework/config_management/child_conf_mgmt_plugin.robot

```diff
@@ -0,0 +1,333 @@
+    Define Child Device name                            #Adding CD in front of the timestamp
+Clean the cloud from devices
+    Remove all managedObjects from cloud                #Removing all existing devices from the tenant 
+Prerequisite Parent
+    Parent Connection                                   #Creates ssh connection to the parent device  
+    ${rc}=    Execute Command    sudo tedge disconnect c8y    return_stdout=False    return_rc=True
+    Should Be Equal    ${rc}    ${0}  
+    
+    Delete child related content                        #Delete any previous created child related configuration files/folders on the parent device
+    Check for child related content                     #Checks if folders that will trigger child device creation are existing
+    Set external MQTT bind address                      #Setting external MQTT bind address which child will use for communication 
+    Set external MQTT port                              #Setting external MQTT port which child will use for communication Default:1883
+    
+    ${rc}=    Execute Command    sudo tedge connect c8y    return_stdout=False    return_rc=True
+    Should Be Equal    ${rc}    ${0} 
+    Restart Configuration plugin                        #Stop and Start c8y-configuration-plugin
+    Close Connection                                    #Closes the connection to the parent device
+Prerequisite Child
+    Child device delete configuration files             #Delete any previous created child related configuration files/folders on the child device
+Prerequisite Cloud
+    GET Parent ID                                       #Get the Parent ID from the cloud
+    GET Parent name                                     #Get the Parent name from the cloud
+Child device bootstrapping
+    Startup child device                                #Setting up/Bootstrapping of a child device
+Get child credentials
+    GET Child ID                                        #Get the Child ID from the cloud
+    GET Child name                                      #Get the Child name from the cloud
+    Validate child Name                                 #This is to check the existence of the bug: https://github.com/thin-edge/thin-edge.io/issues/1569
+Snapshot from device
+    Request snapshot from child device                  #Using the cloud command: "Get snapshot from device"
+    Child device response on snapshot request           #Child device is sending 'executing' and 'successful' MQTT responses
+    No response from child device on snapshot request   #Tests the failing of request after timeout of 10s  
+Child device config update
+    Send configuration to device                        #Using the cloud command: "Send configuration to device"
+    Child device response on update request             #Child device is sending 'executing' and 'successful' MQTT responses
+    No response from child device on config update      #Tests the failing of request after timeout of 10s  
```

Comment:
Since both the positive test case and negative test case are covered here, a few more negative test cases can be added, not necessarily in this PR but as follow-up PRs:

1. Genuine failure use-cases where the child device itself reports a failure
2. An operation timing out even after the child device sends an `executing` response
3. False response from the child device on a `config_snapshot` request where the child device sends a successful response without uploading the config file will still result in a FAILED operation in the cloud.
4. Send a bad URL in a `c8y_DownloadConfigFile` request from the cloud and that response should be a failure as thin-edge won't be able to download that file. In this case, a `tedge/{CHILD_ID}/commands/config_snapshot` request won't even be sent to the child device.

Reply:
Not done in this PR because I need more info about how exactly this can be tested, missing boundary values

Reply:
Sure. Some of these negative test cases are documented in the test plans [here](https://github.com/thin-edge/thin-edge.io/issues/1480#issuecomment-1291522381) and [here](https://github.com/thin-edge/thin-edge.io/issues/1544#issuecomment-1323691550). If those cases need more detailing, just lemme know.